src/command_modules/azure-cli-cloud/azure/cli/command_modules/cloud/custom.py

```diff
@@ -0,0 +1,58 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from azure.cli.core._util import CLIError
+
+from azure.cli.core.cloud import (Cloud,
+                                  get_clouds,
+                                  get_cloud,
+                                  remove_cloud,
+                                  add_cloud,
+                                  CloudAlreadyRegisteredException,
+                                  CloudNotRegisteredException,
+                                  CannotUnregisterCloudException)
+
+def list_clouds():
+    return get_clouds()
+
+def show_cloud(cloud_name):
+    try:
+        return get_cloud(cloud_name)
+    except CloudNotRegisteredException as e:
+        raise CLIError(e)
+
+ # pylint: disable=unused-argument
+def register_cloud(cloud_name,
```

Comment:
We should expose a "discovery_endpoint" or similar thing that if specified would attempt to query the provided URL for all of these settings. Even if not every service or cloud supports it, it would be nice for those that do.